src/main/java/hudson/plugins/scm/koji/KojiChangeLogSet.java

```diff
@@ -83,4 +78,67 @@ public User getAuthor() {
 
     }
 
+    public static class HyperlinkStringContainer {
+
+        private List<HyperlinkString> hyperlinks;
+
+        public HyperlinkStringContainer(String string) {
+            hyperlinks = new ArrayList();
+            hyperlinks.add(new HyperlinkString(string));
+        }
+
+        public HyperlinkStringContainer(List<RPM> rpms) {
+            this.hyperlinks = new ArrayList();
+            storeRpms(rpms);
+        }
+
+        public List<HyperlinkString> getHyperlinks() {
+            return hyperlinks;
+        }
+
+        private void storeRpms(List<RPM> rpms) {
+            for (RPM rpm : rpms) {
+                if (rpm.hasUrl()) {
+                    hyperlinks.add(new HyperlinkString(rpm.toString(), rpm.getUrl()));
+                }
+            }
+        }
+
+        @Override
+        public String toString() {
+            String string = "";
+            for (HyperlinkString s : hyperlinks) {
+                string += s.getString() + ", ";
```

Comment:
Bad idea to concatenate strings in loop. It's performance killer. Even if you know that you will not have many values, you should use StringBuilder. Or maybe use String.join https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.Iterable-

Reply:
anyway, if you properly override `toString` method in `HyperlinkString`(what you should do anyway), it should be enough to call `hyperlinks.toString()`